src/ampycloud/layer.py

```diff
@@ -240,8 +241,27 @@ def ncomp_from_gmm(vals: np.ndarray,
         return best_ncomp, best_ids, abics
 
     # If I found more than one component, let's make sure that they are sufficiently far apart.
-    # First, let's compute the mean component heights
-    mean_comp_heights = [np.mean(vals_orig[best_ids == i]) for i in range(ncomp[best_model_ind])]
+    # First, let's compute the component base altitude
+    def _calc_base_alt(vals, lookback_perc, alt_perc):
+        n_largest_idxs = vals.argsort()[
+            - int(len(vals) * lookback_perc / 100):
+        ]
+        n_largest_elements = vals[n_largest_idxs]
+        if len(n_largest_elements) == 0:
+            warnings.warn(
+                'Cloud base calculation got an empty array.'
+                'Maybe check lookback percentage (is set to %i)' %lookback_perc
+            )
+            return np.nan
```

Comment:
@fpavogt the quantile function raised when I passed it an empty array. The `np.mean` we used before would return a nan and I assume this is what happened before my change. But two things are unclear to me:
1.Is it acceptable to return a nan as cloud base height for a given layer? Do we treat that case somewhere (metarize maybe)?
2. How can we have an empty array here? Seems 🐟 -y no?

Reply:
I should add, the test that failed was `test_identical_pts()` in `test_layers.py` and therein when we have two distinct but close points.

Reply:
Agreed, I would not expect an empty array to be possible. I'll checkout your branch and take a close look. One option is that "non-detection hits" (i.e. hits with type 0 and a NaN altitude) are included somehow ? Just a thought ?

Reply:
I will quickly check if my branch introduced the empty array or if it was there before but went through the `np.mean` unnoticed. That could allow us to pinpoint the issue better.

Reply:
Ok, here is what happens. The empty layer arises in this test because we feed a list of hits with only two distinct values, but search for 3 Gaussian components, which somehow ends up being the preferred amount of components returned by the GMM. 

In operations, this would never happen because after #78, a safety check was included in data.py (inside find_layers() on l.755+). 

My proposition would therefore be the following:
- on the new _calc_base_height(), keep your check for len(n_largest_element) == 0, but raise an error instead of a warning if it happens, because a NaN makes no sense.
- in ncomp_from_gmm(), around l.194, I would add a check for the number of individual values, and limit ncomp_max to this (directly after the check if there's only 1 value):

```
    elif len(np.unique(vals_orig)) < ncomp_max:
        ncomp_max = len(np.unique(vals_orig))
        logger.warning(f'Restricting ncomp_max to the max number of individual values: {ncomp_max}')
```

- I would not change data.py and keep the existing ncomp test in there